src/main/java/net/minestom/server/event/server/ServerListPingEvent.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    // Shortcut Methods
+    /**
+     * Sets the version name for the response.
+     *
+     * @param version The version name for the response data.
+     */
+    public void setVersion(String version) {
+        responseData.setVersion(version);
+    }
+
+    /**
+     * Sets the response protocol version.
+     *
+     * @param protocol The protocol version for the response data.
+     */
+    public void setProtocol(int protocol) {
+        responseData.setProtocol(protocol);
+    }
+
+    /**
+     * Sets the response maximum player count.
+     *
+     * @param maxPlayer The maximum player count for the response data.
+     */
+    public void setMaxPlayer(int maxPlayer) {
+        responseData.setMaxPlayer(maxPlayer);
+    }
+
+    /**
+     * Sets the response online count.
+     *
+     * @param online The online count for the response data.
+     */
+    public void setOnline(int online) {
+        responseData.setOnline(online);
+    }
+
+    /**
+     * Adds some players to the response.
+     *
+     * @param players the players
+     */
+    public void addPlayer(Iterable<Player> players) {
+        responseData.addPlayer(players);
+    }
+
+    /**
+     * Adds a player to the response.
+     *
+     * @param player the player
+     */
+    public void addPlayer(Player player) {
+        addPlayer(player.getUsername(), player.getUuid());
+    }
+
+    /**
+     * Adds a player to the response.
+     *
+     * @param name The name of the player.
+     * @param uuid The unique identifier of the player.
+     */
+    public void addPlayer(String name, UUID uuid) {
+        responseData.addPlayer(name, uuid);
+    }
+
+    /**
+     * Adds a player to the response.
+     * {@link UUID#randomUUID()} is used as a default parameter for uuid
+     */
+    public void addPlayer(String name) {
+        responseData.addPlayer(name, UUID.randomUUID());
+    }
+
+    /**
+     * Removes all of the ping players from this {@link #responseData#pingPlayers}. The {@link #responseData#pingPlayers} list
+     * will be empty this call returns.
+     */
+    public void clearPlayers() {
+        responseData.clearPlayers();
+    }
+
+    /**
+     * Sets the response description.
+     *
+     * @param description The description for the response data.
+     */
+    public void setDescription(Component description) {
+        responseData.setDescription(description);
+    }
+
+    /**
+     * Sets the response favicon.
+     *
+     * MUST start with "data:image/png;base64,"
+     *
+     * @param favicon The favicon for the response data.
+     */
+    public void setFavicon(String favicon) {
+        responseData.setFavicon(favicon);
+    }
+
+    /**
+     * Get the server address a client used to connect.
+     * may be null
+     * @return the server address
+     */
+    public @Nullable String getClientServerAddress() {
+        return Objects.requireNonNull(connection.getHandshakeData()).getServerAddress();
```

Comment:
If the handshake is null, this will throw an exception - is that what we want here?